File: source/frontend/src/app/entities/werkzoekende/mPWerkzoekende.ts
import {Beroepsnaam} from '../shared/beroepsnaam';
import {Contractvorm} from '../shared/contractvorm';
import {Flexibiliteit} from '../shared/flexibiliteit';
import {MPArbeidsmarktkwalificatie} from './mPArbeidsmarktkwalificatie';
import {MPVervoermiddel} from '../shared/mPVervoermiddel';
import {Mobiliteit} from './mobiliteit';
import {SectorBeroepsEnBedrijfsleven} from '../shared/sectorBeroepsEnBedrijfsleven';
import {Voorkeursland} from './voorkeursland';
import {Werktijden} from './werktijden';

export interface MPWerkzoekende {
  indicatieLdrRegistratie?: number;
  indicatieBeschikbaarheidContactgegevens?: number;
  voorkeursland?: Array<Voorkeursland>;
  vervoermiddel?: Array<MPVervoermiddel>;
  mobiliteit?: Mobiliteit;
  flexibiliteit?: Flexibiliteit;
  werktijden?: Werktijden;
  contractvorm?: Array<Contractvorm>;
  arbeidsmarktkwalificatie?: MPArbeidsmarktkwalificatie;
  sector?: Array<SectorBeroepsEnBedrijfsleven>;
  bemiddelingsberoep?: Array<Beroepsnaam>;
}

File: source/backend-werkzoekendenprofielen-bron/src/main/java/com/cap/pocvng/entity/MPWerkzoekende.java
package com.cap.pocvng.entity;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.validation.Valid;
import javax.validation.constraints.Max;
import javax.validation.constraints.Min;
import java.util.List;

/**
 * MPWerkzoekende
 */

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class MPWerkzoekende {

    @Min(1)
    @Max(2)
    private Integer indicatieLdrRegistratie;

    @Min(1)
    @Max(2)
    private Integer indicatieBeschikbaarheidContactgegevens;

    @Valid
    private List<Voorkeursland> voorkeursland;

    @Valid
    private List<MPVervoermiddel> vervoermiddel;

    @Valid
    private Mobiliteit mobiliteit;

    @Valid
    private Flexibiliteit flexibiliteit;

    @Valid
    private Werktijden werktijden;

    @Valid
    private List<Contractvorm> contractvorm;

    @Valid
    private MPArbeidsmarktkwalificatie arbeidsmarktkwalificatie;

    @Valid
    private List<SectorBeroepsEnBedrijfsleven> sector;

    @Valid
    private List<Beroepsnaam> bemiddelingsberoep;

}
